src/app-gocardless/services/gocardless-service.js

```diff
@@ -18,11 +18,22 @@ import { SecretName, secretsService } from '../../services/secrets-service.js';
 
 const GoCardlessClient = nordigenNode.default;
 
-const getGocardlessClient = () =>
-  new GoCardlessClient({
+const clients = new Map();
```

Comment:
Can different budgets have different Nordigen secrets? I guess the `Map` makes sense here 👍

Reply:
Different budgets can have different secrets, but also the same budget might change its keys. So in that case too we also want to generate a new client.